### PR TITLE
Changed the implementation of ThreadTransactionManager

### PR DESCRIPTION
**Pull request by jimfulton**
Changed the implementation of `ThreadTransactionManager` to be a
`thread.local` that wraps a `TransactionManager` rather than a
`thread.local` that inherits from `TransactionManager`.  It now
exposes a `manager` attribute that allows access to the wrapped
transaction manager to allow cross thread calls.  ZODB will use this
to call `unregisterSynch` when a connection is closed from a main
thread.

This is needed to fix the issue https://github.com/zopefoundation/ZODB/issues/208

### CODE REVIEW

File: transaction/_manager.py [hunk 1]
```diff
@@ -215,12 +215,73 @@ def run(self, func=None, tries=3):
...
+    def explicit(self, v):
+        self.manager.explicit = v
+
+    def begin(self):
+        return self.manager.begin()
+
+    def get(self):
+        return self.manager.get()
+
+    __enter__ = get
```
Comment (mgedmin) [inline transaction/_manager.py]: This makes assumptions about TransactionManager's implementation that may be true today, but might change in the future?

Why not forward to `self.manager.__enter__()`?

Comment (jimfulton) [reply transaction/_manager.py]: Fair enough.  I'll change.

**Comment by jimfulton** [comment]
I'll add interface tests.

Thanks for the review!

File: transaction/_manager.py (hunk 1 above)
Comment (jamadden) [reply transaction/_manager.py]: Other than the dunder methods, could this be implemented with `__getattr__` and `__setattr__` forwarding to `self.manager`? That's less explicit, but much shorter. Or does that break interface validation?

Comment (jimfulton) [reply transaction/_manager.py]: done

**Comment by jimfulton** [comment]
Added interface verification tests for managers

**Comment by icemac** [comment]
@jimfulton Is this PR ready to merge? I'd like to include the fix for zopefoundation/ZODB#208 into the next Zope release. (Is there anything else to be done to fix that issue?)
The Plone developers want the Zope 4b7 release before the conference in Tokyo to be able to make a Plone alpha release based on Python 3.

**Comment by jimfulton** [comment]
Sorry, I'm just terribly time challenged.  It would be great if someone else had time to make a release.

**Comment by jimfulton** [comment]
Thanks! And thanks for the release note!

**Pull request merged**